### PR TITLE
ray summit 2024

### PR DESCRIPTION
**Pull request by nnshah1**
Updating tutorial for stable diffusion and ray serve to latest versions.
Added support for launching grafana and prometheus to ray serve example.
Added support for auto scaling based on 1 request per worker (up to 8 workers)

### CODE REVIEW

File: Triton_Inference_Server_Python_API/examples/rayserve/tritonserver_deployment.py [hunk 1]
```diff
@@ -51,18 +51,50 @@ def _print_heading(message):
...
+        if use_torch_compile:
+            print("compiling")
+            print(torch._dynamo.list_backends())
+            self._pipeline.unet = torch.compile(
+                self._pipeline.unet,
+                fullgraph=True,
+                mode="reduce-overhead",
+                dynamic=False,
+            )
+        self.generate("temp")
```
Comment (rmccorm4) [inline Triton_Inference_Server_Python_API/examples/rayserve/tritonserver_deployment.py:97]: Is this warmup or something?

File: Triton_Inference_Server_Python_API/build.sh [hunk 2]
```diff
@@ -39,8 +39,8 @@ DOCKERFILE=${SOURCE_DIR}/docker/Dockerfile
 
 # Base Images
 BASE_IMAGE=nvcr.io/nvidia/tritonserver
-BASE_IMAGE_TAG_IDENTITY=24.01-py3
-BASE_IMAGE_TAG_DIFFUSION=24.01-py3
+BASE_IMAGE_TAG_IDENTITY=24.08-py3
+BASE_IMAGE_TAG_DIFFUSION=24.08-py3
 BASE_IMAGE_TAG_TRT_LLM=24.01-trtllm-python-py3
```
Comment (rmccorm4) [inline Triton_Inference_Server_Python_API/build.sh:44]: Does TRT-LLM image need update to 24.08?

File: Popular_Models_Guide/StableDiffusion/build.sh
Comment (rmccorm4) [inline Popular_Models_Guide/StableDiffusion/build.sh:1]: I think the README needs some version updates too for 24.08 and TRT 10.4 where applicable:

ex:

> The example is designed based on the nvcr.io/nvidia/tritonserver:24.01-py3 docker image and [TensorRT OSS v9.2.0](https://github.com/NVIDIA/TensorRT/releases/tag/v9.2.0).

**Review by rmccorm4 — COMMENTED** [review]
LGTM other than current comments

File: Triton_Inference_Server_Python_API/examples/rayserve/tritonserver_deployment.py (hunk 1 above)
Comment (nnshah1) [reply Triton_Inference_Server_Python_API/examples/rayserve/tritonserver_deployment.py:97]: yeah - for the torch compile version - I found that the compilation didn't happen until the first pipeline invocation - so wanted to do that during initialization - will double check if that is still the case

**Review by rmccorm4 — APPROVED** [review]
🚀

**Pull request merged**